test/integration/authorise_application_test.rb

```diff
@@ -85,6 +86,10 @@ def assert_access_granted(user, app)
     assert_kind_of Doorkeeper::AccessGrant, Doorkeeper::AccessGrant.find_by(resource_owner_id: user.id, application: app)
   end
 
+  def refute_access_granted(user)
```

Comment:
Minor: I know rubocop has issues with using "refute" in tests, preferring "assert_not". Not sure I agree with the cop, but it might be worth calling this method `assert_not_access_granted` to avoid confusion?

Reply:
I've addressed this in 2b1b137468d4d9f5e892486d8b22da3f64f21a18.